server/src/test/kotlin/tech/figure/objectstore/gateway/server/ObjectStoreGatewayServerTest.kt

```diff
@@ -295,6 +300,148 @@ class ObjectStoreGatewayServerTest {
+    fun `batchGrantScopePermission should respond with all successes to an authorized caller for multiple grantees with grant ids`() {
+        testBatchGrantScopePermission(
+            contextKeyPair = keyPair,
+            *(0..10).map { grantIdPrefix -> buildScopeGrantee(grantId = "$grantIdPrefix-${UUID.randomUUID()}") }.toTypedArray()
+        )
+    }
+
+    @Test
+    fun `batchGrantScopePermission should reject grants that are not authorized`() {
+        // Use some random account as the requesting account to verify that the requester has to be someone relevant
+        val granterAccount = genRandomAccount()
+        setUpBaseServices(contextKeyPair = granterAccount.keyPair.toJavaECKeyPair())
+        setUpScopePermissionValues()
+        val responseObserver = mockkObserver<BatchGrantScopePermissionResponse>()
+        val grantees = (0..10).map { buildScopeGrantee() }
+        val request = getBatchGrant(*grantees.toTypedArray())
+        server.batchGrantScopePermission(request = request, responseObserver = responseObserver)
+        verify(inverse = true) { responseObserver.onError(any()) }
+        verifyAll {
+            responseObserver.onNext(
+                BatchGrantScopePermissionResponse.newBuilder().also { response ->
+                    response.request = request
+                    response.addAllGrantResponses(
+                        grantees.map { grantee ->
+                            GrantScopePermissionResponse.newBuilder().also { grantResponse ->
+                                grantResponse.requestBuilder.scopeAddress = scopeAddress
+                                grantResponse.requestBuilder.granteeAddress = grantee.granteeAddress
+                                grantResponse.grantAccepted = false
+                            }.build()
+                        }
+                    )
+                }.build()
+            )
+            responseObserver.onCompleted()
+        }
+        assertEquals(
+            expected = 0,
+            actual = transaction { ScopePermissionsTable.selectAll().count() },
+            message = "No scope permissions should have ever been assigned, proving that all grantees were rejected",
+        )
+    }
+
+    @Test
+    fun `batchGrantScopePermission should remove grants that cause exceptions from the response`() {
+        setUpBaseServices()
+        setUpScopePermissionValues()
+        val goodGrantee = buildScopeGrantee()
+        val badGrantee = buildScopeGrantee()
+        // Hook into the database to force it to throw an exception only on this address
+        every {
+            scopePermissionsRepository.addAccessPermission(
+                scopeAddress = scopeAddress,
+                granteeAddress = badGrantee.granteeAddress,
+                granterAddress = defaultGranter,
+                grantId = null,
+            )
+        } throws IllegalStateException("For some reason, this particular address causes exceptions in the database!")
```

Comment:
haha, I was wondering what case would cause a grantee to be rejected... I guess this shows we have no clue 😆

Reply:
Truth lol.  I was like "what kinda thing can I hook into here to simulate an error" and the choices were VERY slim lol